app/models/manageiq/providers/hawkular/inventory/parser/middleware_manager.rb

```diff
@@ -332,12 +332,21 @@ def parse_middleware_server(eap, inventory_object, domain = false, name = nil)
           not_started && eap.properties[x].nil? ? _('not yet available') : eap.properties[x]
         end
 
-        inventory_object.assign_attributes(
+        attributes = {
           :name      => name || parse_standalone_server_name(eap.id),
           :type_path => eap.type_path,
           :hostname  => hostname,
           :product   => product
-        )
+        }
+
+        case product
+        when /wildfly/i
+          attributes[:type] = 'ManageIQ::Providers::Hawkular::MiddlewareManager::MiddlewareServerWildfly'
+        when /eap/i
+          attributes[:type] = 'ManageIQ::Providers::Hawkular::MiddlewareManager::MiddlewareServerEap'
+        end
```

Comment:
- Minor, but you can extract the commonailty in the two branches like so:

```ruby
attributes[:type] =
  case product
  when /wildfly/i then 'ManageIQ::Providers::Hawkular::MiddlewareManager::MiddlewareServerWildfly'
  when /eap/i    then 'ManageIQ::Providers::Hawkular::MiddlewareManager::MiddlewareServerEap'
  end

- What happens if the product is *neither* of those two cases for some reason.  Should you raise an exception?

Reply:
I don't think that's needed, since we're 'just' selecting a more specific type. If neither case is matched, the `type` is left unchanged, which means the value is `ManageIQ::Providers::Hawkular::MiddlewareManager::MiddlewareServer`